src/system/sargon_os/sargon_os_security_structures.rs

```diff
@@ -0,0 +1,306 @@
+use crate::prelude::*;
+
+#[uniffi::export]
+impl SargonOS {
+    /// Returns all the SecurityStructuresOfFactorSources,
+    /// by trying to map FactorSourceID level -> FactorSource Level
+    pub fn security_structures_of_factor_sources(
+        &self,
+    ) -> Result<SecurityStructuresOfFactorSources> {
+        self.profile_holder
+            .access_profile_with(|p| p.security_structures_of_factor_sources())
+    }
+
+    /// Returns all the `SecurityStructuresOfFactorSourceIDs` which are stored
+    /// in profile.
+    pub fn security_structures_of_factor_source_ids(
+        &self,
+    ) -> SecurityStructuresOfFactorSourceIDs {
+        self.profile_holder.access_profile_with(|p| {
+            p.app_preferences
+                .security
+                .security_structures_of_factor_source_ids
+                .clone()
+        })
+    }
+
+    /// Returns all the `SecurityStructuresOfFactorSourceIDs` which are stored
+    /// in profile.
+    pub fn security_structure_of_factor_sources_from_security_structure_of_factor_source_ids(
+        &self,
+        structure_of_ids: &SecurityStructureOfFactorSourceIDs,
+    ) -> Result<SecurityStructureOfFactorSources> {
+        self.profile_holder.try_access_profile_with(|p| {
+            SecurityStructureOfFactorSources::try_from((
+                structure_of_ids,
+                &p.factor_sources,
+            ))
+        })
+    }
+
+    /// Adds the security stricture of factor sources to Profile if none with the
```

Comment:
typo structure

Reply:
why did not `typos` CLI tool pick this up?!

Reply:
![Screenshot 2024-06-14 at 10 11 06](https://github.com/radixdlt/sargon/assets/164921079/82c2f523-875f-4a8c-bf54-f2c9a01de497)

Reply:
aha!